### DISCUSSION TITLE
Matching an expression and pushing a different one

### DISCUSSION THREAD

**Discussion by jhoobergs** [pest-parser/pest#880] [Q&A]
I am writing a parser for a subset of yaml (https://github.com/m8rex/yaml-subset) and I need to be able to match for a `-` but push a a space (` `) to the stack.

The reason for this is that if you place a hash in a list in yaml, you get something like this:
```yaml
---
field:
  - type: item
    other: value
```

To match the indentation of the inner hash, I would need to push a space onto the stack, when matching the `-` of the list.

I would like something like `PUSH_OTHER("-", " ")`.

I looked into the code quickly, but it seems that the pushes happen with span's into the source text, which does not make it easy to push something else. Any thoughts?

**Comment by tomtau** [comment]
A side note: I saw there was this yaml grammar: https://github.com/aofdev/yaml-peg/blob/main/src/parser/grammar/yaml.pest but not sure whether it's complete / correct

**Reply by jhoobergs** [reply]
The main difference is that I want to preserve the comments, which is rater hard in yaml, but needed for formatting purposes. I also don't think that it is complete.

**Comment by yshavit** [comment]
Would there be any appetite for doing this, if I were to look into creating a PR? Or is it something this project actively wants to avoid?

Another example of when it would be useful: a rule that starts with either `[` or `(`, and then ends with `]` or `)` respectively.

**Comment by yshavit** [comment]
This has been completed as of #1092. The syntax is:

```pest
PUSH_LITERAL( <literal string> )
```

This is slightly different from the proposed `PUSH_OTHER`, in that it only takes one argument, and always matches. You can get the equivalent of `PUSH_OTHER` by combining this with a sequence match:

```pest
"-" ~ PUSH_LITERAL(" ")
```
